realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/CredentialsTests.kt

```diff
@@ -151,33 +176,43 @@ class CredentialsTests {
         assertFailsWith<IllegalArgumentException> { Credentials.jwt(TestHelper.getNull()) }
     }
 
-    fun expectErrorCode(app: App, expectedCode: ErrorCode, credentials: Credentials) {
-        try {
-            app.login(credentials)
-            fail()
-        } catch (error: AppException) {
-            assertEquals(expectedCode, error.errorCode)
-        }
-    }
-
     @Test
     fun loginUsingCredentials() {
         app = TestApp()
+        admin = ServerAdmin()
+
         Credentials.IdentityProvider.values().forEach { provider ->
-            when(provider) {
+            when (provider) {
                 Credentials.IdentityProvider.ANONYMOUS -> {
                     val user = app.login(Credentials.anonymous())
                     assertNotNull(user)
                 }
                 Credentials.IdentityProvider.API_KEY -> {
-                    // FIXME: Wait for API Key support in OS
-//                        val user: User = app.registerUserAndLogin(TestHelper.getRandomEmail(), "123456")
-//                        val key: UserApiKey = app.apiKeyAuthProvider.createApiKey("my-key");
-//                        val apiKeyUser = app.login(Credentials.apiKey(key.value!!))
-//                        assertNotNull(apiKeyUser)
+                    // Log in, create an API key, log out, log in with the key, compare users
+                    val user: User = app.registerUserAndLogin(TestHelper.getRandomEmail(), "123456")
+                    val key: UserApiKey = user.apiKeyAuth.createApiKey("my-key");
+                    user.logOut()
+                    val apiKeyUser = app.login(Credentials.apiKey(key.value!!))
+                    assertEquals(user, apiKeyUser)
```

Comment:
Not sure how our equals method works, but it seems a bit weird that a logged out user is equals to a logged in user... So we probably need to revist that equals method... Comparing the user id should work though